File: Source/Aladdin.CAPI.GOST.Java/src/aladdin/capi/gost/keys/GOSTR3412.java
package aladdin.capi.gost.keys;
import aladdin.capi.*;

///////////////////////////////////////////////////////////////////////////
// Ключ ГОСТ Р34.12
///////////////////////////////////////////////////////////////////////////
public class GOSTR3412 extends SecretKeyFactory
{
    // тип ключа
    public static final SecretKeyFactory INSTANCE = new GOSTR3412(); 
        
    // конструктор
    private GOSTR3412() { super("GOSTR3412"); }
    
	// размер ключей
	@Override public final int[] keySizes() { return new int[] { 32 }; }
}

File: Source/Aladdin.CAPI.GOST/CAPI/GOST/Keys/GOSTR3412.cs
namespace Aladdin.CAPI.GOST.Keys
{
    ///////////////////////////////////////////////////////////////////////////
    // Ключ ГОСТ Р34.12
    ///////////////////////////////////////////////////////////////////////////
    public class GOSTR3412 : SecretKeyFactory
    {
        // тип ключа
        public static readonly SecretKeyFactory Instance = new GOSTR3412(); 

        // размер ключей
	    public override int[] KeySizes { get { return new int[] { 32 }; }}
    }
}
